File: TF17. PHP MVC AND SYMFONY OVERVIEW - EX (CALCULATOR)/src/CalculatorBundle/Entity/Calculator.php
<?php

namespace CalculatorBundle\Entity;

use Doctrine\ORM\Query\Expr\Math;

class Calculator
{
    /**
     * @var float
     */
    private $leftOperand;
    /**
     * @var float
     */
    private $rightOperand;

    /**
     * @var string
     */
    private $operator;

    /**
     * @return float
     */
    public function getLeftOperand(): ?float
    {
        return $this->leftOperand;
    }

    /**
     * @param float $leftOperand
     */
    public function setLeftOperand(float $leftOperand)
    {
        $this->leftOperand = $leftOperand;
    }

    /**
     * @return float
     */
    public function getRightOperand(): ?float
    {
        return $this->rightOperand;
    }

    /**
     * @param float $rightOperand
     */
    public function setRightOperand(float $rightOperand)
    {
        $this->rightOperand = $rightOperand;
    }

    /**
     * @return string
     */
    public function getOperator(): ?string
    {
        return $this->operator;
    }

    /**
     * @param string $operator
     */
    public function setOperator(string $operator)
    {
        $this->operator = $operator;
    }

    /**
     * @return float
     */
    public function calculateResult(): ?float
    {
        $result = 0;

        switch ($this->getOperator()) {

            case "+":
                $result = $this->getLeftOperand() + $this->getRightOperand();
                break;
            case "-":
                $result = $this->getLeftOperand() - $this->getRightOperand();
                break;
            case "*":
                $result = $this->getLeftOperand() * $this->getRightOperand();
                break;
            case "/":
                $result = $this->getLeftOperand() / $this->getRightOperand();
                break;
            case "^":
                $result = pow($this->getLeftOperand(), $this->getRightOperand());
                break;
            case "√":
                $result = sqrt($this->getLeftOperand());
                break;
            case "mod":
                $result = $this->getLeftOperand() % $this->getRightOperand();
                break;
            case "sin":
                $result = number_format(sin($this->getLeftOperand() / (180 / pi())), 6, '.', '');
                break;
            case "cos":
                $result = number_format(cos($this->getLeftOperand() / (180 / pi())), 6, '.', '');
                break;
            case "tan":
                $result = number_format(tan($this->getLeftOperand() / (180 / pi())), 6, '.', '');
                break;
            case "!":
                function factorial($num)
                {
                    if ($num < 0) {
                        return NAN;
                    } else if ($num == 0) {
                        return '1';
                    } else {
                        return ($num * factorial($num - 1));
                    }
                }

                $result = factorial($this->getLeftOperand());
                break;
            case "ln":
                $result = number_format(log($this->getLeftOperand()), 6, '.', '');
                break;
        }

        return $result;
    }
}

File: TF23. JAVA BASIC WEB - EX (CALCULATOR)/src/main/java/com/softuni/entity/Calculator.java
package com.softuni.entity;

public class Calculator {
    private double leftOperand;
    private double rightOperand;
    private String operator;

    public Calculator(double leftOperand, double rightOperand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOperand = rightOperand;
        this.operator = operator;
    }

    public double getLeftOperand() {
        return leftOperand;
    }

    public void setLeftOperand(double leftOperand) {
        this.leftOperand = leftOperand;
    }

    public double getRightOperand() {
        return rightOperand;
    }

    public void setRightOperand(double rightOperand) {
        this.rightOperand = rightOperand;
    }

    public String getOperator() {
        return operator;
    }

    public void setOperator(String operator) {
        this.operator = operator;
    }

    public double calculateResult() {
        double result = 0.0;

        switch (this.operator) {
            case "+": {
                result = this.leftOperand + this.rightOperand;
            }
            break;
            case "-": {
                result = this.leftOperand - this.rightOperand;
            }
            break;
            case "*": {
                result = this.leftOperand * this.rightOperand;
            }
            break;
            case "/": {
                result = this.leftOperand / this.rightOperand;
            }
            break;
            case "^": {
                result = Math.pow(this.leftOperand, this.rightOperand);
            }
            break;
            case "mod": {
                result = this.leftOperand % this.rightOperand;
            }
            break;
            case "√": {
                result = Math.sqrt(this.leftOperand);
            }
            break;
            case "sin": {
                result = Math.sin(this.leftOperand / (180 / Math.PI));
            }
            break;
            case "cos": {
                result = Math.cos(this.leftOperand / (180 / Math.PI));
            }
            break;
            case "tan": {
                result = Math.tan(this.leftOperand / (180 / Math.PI));
            }
            break;
            case "!": {
                result = factorial(this.leftOperand);
            }
            break;
            case "ln": {
                result = Math.log(this.leftOperand);
            }
            break;

            default: {
                result = 0;
            }
            break;
        }
        return result;
    }


    private double factorial(double num) {
        if (num < 0) {
            return Double.NaN;
        } else if (num == 0) {
            return 1;
        } else {
            return (num * factorial(num - 1));
        }
    }
}